connect/transforms/src/main/java/org/apache/kafka/connect/transforms/MaskField.java

```diff
@@ -99,14 +119,23 @@ private R applyWithSchema(R record) {
         final Struct updatedValue = new Struct(value.schema());
         for (Field field : value.schema().fields()) {
             final Object origFieldValue = value.get(field);
-            updatedValue.put(field, maskedFields.contains(field.name()) ? masked(origFieldValue) : origFieldValue);
+            updatedValue.put(field, maskedFields.contains(field.name())
+                    ? masked(origFieldValue)
+                    : origFieldValue);
         }
         return newRecord(record, updatedValue);
     }
 
-    private static Object masked(Object value) {
-        if (value == null)
+
+    private Object masked(Object value) {
+        if (value == null) {
             return null;
+        }
+
+        if (replacement != null && !replacement.isEmpty()) {
```

Comment:
Is it valid to have a blank string as the replacement value? If not, then the `replacement` config should have a validator that prevents using invalid values, and it probably would be good to succinctly describe the limitations in the doc string.

And it may be better to only set `this.replacement` to a non-null string that should always be used. This would centralize the logic of determining whether it should be used in one place, and line 135 becomes a lot simpler and more efficient:

```suggestion
        if (replacement != null) {
```